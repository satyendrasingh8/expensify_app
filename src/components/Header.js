import React from "react";
import  { NavLink} from 'react-router-dom';

            const Header=() => (
                <header>
                <h1>Expensify </h1>
                <ul> 
               <li> <NavLink to="/" activeClassName="is-active" exact={true}> Dasboard page </NavLink></li>
               <li> <NavLink to="/create" activeClassName="is-active"> create page </NavLink></li>
             
               <li> <NavLink to="/help" activeClassName="is-active"> help page </NavLink></li>
             </ul>
                </header>
            );
                export default Header;