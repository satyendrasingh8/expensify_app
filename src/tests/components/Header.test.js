import React from 'react';
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';

test('test Header',() => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot()
    // expect(wrapper.find('h1').text()).toBe('Expensify ');
//const renderer =new ReactShallowRenderer();
 //renderer.render(<Header />);
 //expect(renderer.getRenderOutput()).toMatchSnapshot();
 
})