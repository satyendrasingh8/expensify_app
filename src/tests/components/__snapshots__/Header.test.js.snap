// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify 
</h1>,
        <ul>
          <li>
                     
                    <Unknown
                              activeClassName="is-active"
                              exact={true}
                              to="/"
                    >
                               Dasboard page 
                    </Unknown>
          </li>
          <li>
                     
                    <Unknown
                              activeClassName="is-active"
                              to="/create"
                    >
                               create page 
                    </Unknown>
          </li>
          <li>
                     
                    <Unknown
                              activeClassName="is-active"
                              to="/help"
                    >
                               help page 
                    </Unknown>
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify ",
        },
        "ref": null,
        "rendered": "Expensify ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
               
              <Unknown
                            activeClassName="is-active"
                            exact={true}
                            to="/"
              >
                             Dasboard page 
              </Unknown>
</li>,
            <li>
               
              <Unknown
                            activeClassName="is-active"
                            to="/create"
              >
                             create page 
              </Unknown>
</li>,
            <li>
               
              <Unknown
                            activeClassName="is-active"
                            to="/help"
              >
                             help page 
              </Unknown>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <Unknown
                  activeClassName="is-active"
                  exact={true}
                  to="/"
>
                   Dasboard page 
</Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "children": " Dasboard page ",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": " Dasboard page ",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "NavLink",
                  "propTypes": Object {
                    "activeClassName": [Function],
                    "activeStyle": [Function],
                    "aria-current": [Function],
                    "className": [Function],
                    "exact": [Function],
                    "innerRef": [Function],
                    "isActive": [Function],
                    "location": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "strict": [Function],
                    "style": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <Unknown
                  activeClassName="is-active"
                  to="/create"
>
                   create page 
</Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "children": " create page ",
                  "to": "/create",
                },
                "ref": null,
                "rendered": " create page ",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "NavLink",
                  "propTypes": Object {
                    "activeClassName": [Function],
                    "activeStyle": [Function],
                    "aria-current": [Function],
                    "className": [Function],
                    "exact": [Function],
                    "innerRef": [Function],
                    "isActive": [Function],
                    "location": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "strict": [Function],
                    "style": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <Unknown
                  activeClassName="is-active"
                  to="/help"
>
                   help page 
</Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "is-active",
                  "children": " help page ",
                  "to": "/help",
                },
                "ref": null,
                "rendered": " help page ",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "NavLink",
                  "propTypes": Object {
                    "activeClassName": [Function],
                    "activeStyle": [Function],
                    "aria-current": [Function],
                    "className": [Function],
                    "exact": [Function],
                    "innerRef": [Function],
                    "isActive": [Function],
                    "location": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "strict": [Function],
                    "style": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify 
</h1>,
          <ul>
            <li>
                         
                        <Unknown
                                    activeClassName="is-active"
                                    exact={true}
                                    to="/"
                        >
                                     Dasboard page 
                        </Unknown>
            </li>
            <li>
                         
                        <Unknown
                                    activeClassName="is-active"
                                    to="/create"
                        >
                                     create page 
                        </Unknown>
            </li>
            <li>
                         
                        <Unknown
                                    activeClassName="is-active"
                                    to="/help"
                        >
                                     help page 
                        </Unknown>
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify ",
          },
          "ref": null,
          "rendered": "Expensify ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                <Unknown
                                activeClassName="is-active"
                                exact={true}
                                to="/"
                >
                                 Dasboard page 
                </Unknown>
</li>,
              <li>
                 
                <Unknown
                                activeClassName="is-active"
                                to="/create"
                >
                                 create page 
                </Unknown>
</li>,
              <li>
                 
                <Unknown
                                activeClassName="is-active"
                                to="/help"
                >
                                 help page 
                </Unknown>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <Unknown
                    activeClassName="is-active"
                    exact={true}
                    to="/"
>
                     Dasboard page 
</Unknown>,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "is-active",
                    "children": " Dasboard page ",
                    "exact": true,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": " Dasboard page ",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "NavLink",
                    "propTypes": Object {
                      "activeClassName": [Function],
                      "activeStyle": [Function],
                      "aria-current": [Function],
                      "className": [Function],
                      "exact": [Function],
                      "innerRef": [Function],
                      "isActive": [Function],
                      "location": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "strict": [Function],
                      "style": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  },
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <Unknown
                    activeClassName="is-active"
                    to="/create"
>
                     create page 
</Unknown>,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "is-active",
                    "children": " create page ",
                    "to": "/create",
                  },
                  "ref": null,
                  "rendered": " create page ",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "NavLink",
                    "propTypes": Object {
                      "activeClassName": [Function],
                      "activeStyle": [Function],
                      "aria-current": [Function],
                      "className": [Function],
                      "exact": [Function],
                      "innerRef": [Function],
                      "isActive": [Function],
                      "location": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "strict": [Function],
                      "style": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  },
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <Unknown
                    activeClassName="is-active"
                    to="/help"
>
                     help page 
</Unknown>,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "is-active",
                    "children": " help page ",
                    "to": "/help",
                  },
                  "ref": null,
                  "rendered": " help page ",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "NavLink",
                    "propTypes": Object {
                      "activeClassName": [Function],
                      "activeStyle": [Function],
                      "aria-current": [Function],
                      "className": [Function],
                      "exact": [Function],
                      "innerRef": [Function],
                      "isActive": [Function],
                      "location": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "strict": [Function],
                      "style": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  },
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
